src/test/java/com/example/training/controllers/UserControllerTest.java

```diff
@@ -28,34 +31,49 @@
     private static final String SINGLE_USER_PATH = "src/test/resources/user/singleUser.json";
     private static final User TEST_USER1 = new User("2a3993ee-d74e-42ac-af90-9390642db803", "John", "Doe");
     private static final User TEST_USER2 = new User("8644d523-eaf2-412b-87b9-82ceba72ec6b", "Foo", "Bar");
+    private static final User TEST_USER_TO_ADD = new User("8644d523-eaf2-xxxx-87b9-82ceba72ec6b", "Joe", "Ziggy");
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private MockMvc mockMvc;
 
     @Mock
-    private SimpleListRepository simpleListRepository;
+    private H2Repository h2Repository;
 
     @Before
     public void setup() {
-        this.mockMvc = standaloneSetup(new UserController(simpleListRepository)).build();
+        this.mockMvc = standaloneSetup(new UserController(h2Repository)).build();
 
-        when(simpleListRepository.getAllUsers()).thenReturn(ImmutableList.of(TEST_USER1, TEST_USER2));
-        when(simpleListRepository.getUser(1)).thenReturn(TEST_USER1);
+        when(h2Repository.getAllUsers()).thenReturn(ImmutableList.of(TEST_USER1, TEST_USER2));
+        when(h2Repository.getUserById("1")).thenReturn(TEST_USER1);
     }
 
     @Test
     public void When_GettingAllUsers_Expect_CorrectResponse() throws Exception {
         this.mockMvc.perform(get("/users"))
-            .andExpect(status().isOk())
-            .andExpect(content().json(readJsonFile(ALL_USERS_PATH)));
+                .andExpect(status().isOk())
+                .andExpect(content().json(readJsonFile(ALL_USERS_PATH)));
     }
 
     @Test
     public void When_GettingUserById_Expect_CorrectResponse() throws Exception {
         this.mockMvc.perform(get("/users/1"))
-            .andExpect(status().isOk())
-            .andExpect(content().json(readJsonFile(SINGLE_USER_PATH)));
+                .andExpect(status().isOk())
+                .andExpect(content().json(readJsonFile(SINGLE_USER_PATH)));
+    }
+
+    @Test
+    public void When_addingUser_Expect_CorrectResponse() throws Exception {
```

Comment:
What will happen when you try to add user that already exist?
It should be an update...